bitbucket-pipelines.dist.yml

```diff
@@ -0,0 +1,127 @@
+---
+
+definitions:
+  installscript: &baseInstallDeps
+    export DEBIAN_FRONTEND='noninteractive' ;
+    export COMPOSER_ALLOW_SUPERUSER=1 ;
+    IPADDRESS=$(grep "$HOSTNAME$" /etc/hosts | grep -Po '^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}') ; export IPADDRESS ;
+    apt-get -qq update &&
+    mkdir -p /usr/share/man/man1 /usr/share/man/man3 /usr/share/man/man7 &&
+    apt-get -qq install default-jre-headless postgresql-client mariadb-client &&
+    curl -sS https://raw.githubusercontent.com/creationix/nvm/v0.38.0/install.sh | bash &&
+    . ~/.bashrc &&
+    nvm install -b --no-progress v14.15.0 &&
+    curl -sS https://getcomposer.org/installer | php -- --install-dir='/usr/local/bin' --filename='composer' &&
+    touch /usr/local/bin/phpenv && chmod +x /usr/local/bin/phpenv &&
+    mkdir -pv /tmp/plugin ; cp -ru "${BITBUCKET_CLONE_DIR}"/* /tmp/plugin ; rm -rf "${BITBUCKET_CLONE_DIR:?}"/* ;
+    mkdir -pv "${BITBUCKET_CLONE_DIR}/plugin" && cp -ru /tmp/plugin/* "${BITBUCKET_CLONE_DIR}"/plugin ;
+    composer create-project -n --no-dev --no-progress --no-ansi --prefer-dist moodlehq/moodle-plugin-ci ci ^3 &&
+    { [ "${MOODLE_START_BEHAT_SERVERS}" = '' ] && export MOODLE_START_BEHAT_SERVERS='NO' ; } ;
```

Comment:
Reason for all this is to offer flexibility to the people planning to use this. This script works out of the box as is.
Here is an example repository 
https://bitbucket.org/moodlework/moodle-mod_bigbluebuttonbn/
And pipelines
https://bitbucket.org/moodlework/moodle-mod_bigbluebuttonbn/addon/pipelines/home

The plugin-ci is designed only for travis so any other system has to prepare things before to avoid plugin throwing errors.
I can also provide a bit simplified version of this script but felt some users wanted a complex full blown examples that respond to lot of questions on how to do this.

As for why some variables are defined if not present - because that way one provides defaults. Bitbucket pipelines, unfortunately does not have means of setting pipeline variables like travis or even gitlab do. The only thing left is shell script.